core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/ControlInputPort.scala

```diff
@@ -36,8 +37,10 @@ class ControlInputPort {
   @inline
   private def processControlEvents(iter: Iterable[ControlPayload]): Unit = {
     iter.foreach {
+      case p: PromisePayload =>
+        promiseManager.execute(p)
       case other =>
-      //TODO: implement future/promise here
+      //skip for now
```

Comment:
what are these others? At least we should add a log line here otherwise it's just a source of unknown bugs

Reply:
log line added

Reply:
follow the logging style